examples/ext/cooldowns/custom_bucket.py

```diff
@@ -0,0 +1,23 @@
+from typing import Tuple
+
+import hikari
+
+import crescent
+from crescent.ext import cooldowns
+
+bot = crescent.Bot("...")
+
+
+def guild_specific_rate_limits(ctx: crescent.Context) -> Tuple[hikari.Snowflake, hikari.Snowflake]:
+    return (ctx.user.id, ctx.guild.id)
```

Comment:
ctx.guild is sometimes None. Return type should be `int | None`, should also use `ctx.guild_id`

Reply:
i decided to do `Optional[Snowflake]` because i didn't want to add `from __future__ import annotations`